qiskit/quantum_info/operators/symplectic/clifford.py

```diff
@@ -0,0 +1,420 @@
+
+    def __repr__(self):
+        return 'Clifford({})'.format(repr(self.table))
+
+    def __str__(self):
+        return 'Clifford: Stabilizer = {}, Destabilizer = {}'.format(
+            str(self.stabilizer.to_labels()),
+            str(self.destabilizer.to_labels()))
+
+    def __eq__(self, other):
+        """Check if two Clifford tables are equal"""
+        return super().__eq__(other) and self._table == other._table
+
+    # ---------------------------------------------------------------------
+    # Attributes
+    # ---------------------------------------------------------------------
+    def __getitem__(self, key):
+        """Return a stabilizer Pauli row"""
+        return self._table.__getitem__(key)
+
+    def __setitem__(self, key, value):
+        """Set a stabilizer Pauli row"""
+        self._table.__setitem__(key, value)
+
+    @property
+    def n_qubits(self):
+        """The number of qubits for the Clifford."""
+        return self._table._n_qubits
+
+    @property
+    def table(self):
+        """Return StabilizerTable"""
+        return self._table
+
+    @table.setter
+    def table(self, value):
+        """Set the stabilizer table"""
+        # Note that is setup so it can't change the size of the Clifford
+        # It can only replace the contents of the StabilizerTable with
+        # another StabilizerTable of the same size.
+        if not isinstance(value, StabilizerTable):
+            value = StabilizerTable(value)
+        self._table._array[:, :] = value._table._array
+        self._table._phase[:] = value._table._phase
+
+    @property
+    def stabilizer(self):
+        """Return the stabilizer block of the StabilizerTable."""
+        return StabilizerTable(self._table[self.n_qubits:2*self.n_qubits])
+
+    @stabilizer.setter
+    def stabilizer(self, value):
+        """Set the value of stabilizer block of the StabilizerTable"""
+        inds = slice(self.n_qubits, 2*self.n_qubits)
+        self._table.__setitem__(inds, value)
+
+    @property
+    def destabilizer(self):
+        """Return the destabilizer block of the StabilizerTable."""
+        return StabilizerTable(self._table[0:self.n_qubits])
+
+    @destabilizer.setter
+    def destabilizer(self, value):
+        """Set the value of destabilizer block of the StabilizerTable"""
+        inds = slice(0, self.n_qubits)
+        self._table.__setitem__(inds, value)
+
+    # ---------------------------------------------------------------------
+    # Utility Operator methods
+    # ---------------------------------------------------------------------
+
+    def is_unitary(self, atol=None, rtol=None):
+        """Return True if the Clifford table is valid."""
+        # A valid Clifford is always unitary, so this function is really
+        # checking that the underlying Stabilizer table array is a valid
+        # Clifford array.
+
+        # Condition is
+        # table.T * [[0, 1], [1, 0]] * table = [[0, 1], [1, 0]]
```

Comment:
this actually checks that the clifford table is a symplectic matrix, so perhaps this method should be called 'is_symplectic' ?